src/app/views/query-runner/query-input/AutoComplete.tsx

```diff
@@ -123,20 +132,32 @@ class AutoComplete extends Component<IAutoCompleteProps, IAutoCompleteState> {
   }
 
   private getParameterEnums = (url: string) => {
+    const { autoCompleteOptions, sampleQuery: { selectedVerb } } = this.props;
+    if (!autoCompleteOptions) {
+      return;
+    }
+    const parameters = autoCompleteOptions.parameters;
     const param = url.split('$').pop()!.split('=')[0];
-    const { parameters } = this.props.autoCompleteOptions;
-    const section = parameters.find(k => k.name === `$${param}`);
-    const list: string[] = [];
-    if (section.items.length > 0) {
-      section.items.forEach((element: string) => {
-        list.push(element);
-      });
-      this.setState({
-        filteredSuggestions: list,
-        suggestions: list,
-        showSuggestions: true,
-        compare: ''
-      });
+    if (parameters) {
+      const parametersWithVerb = parameters.find(parameter => parameter.verb === selectedVerb.toLowerCase());
+      if (parametersWithVerb) {
+        const section = parametersWithVerb.values.find((k: { name: string; }) => {
+          return k.name === `$${param}`;
+        });
+
+        const list: string[] = [];
+        if (section && section.items && section.items.length > 0) {
+          section.items.forEach((element: string) => {
```

Comment:
From what I see here. I think this `forEach` loop is not necessary.
Since you are just copying items from `section.items` array, I think you could just change the `const` list variable to a `let` and then assign it to `section.items`  i.e
```
let list: string[] = [];

// the other code

list = section.items;
```
I think this will boost the time complexity here,
what do you think?